sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -0,0 +1,103 @@
+package manager
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/timestamp"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	openNetworkEndpoint = &sensor.NetworkEndpoint{
+		SocketFamily: sensor.SocketFamily_SOCKET_FAMILY_IPV4,
+		Protocol:     storage.L4Protocol_L4_PROTOCOL_TCP,
+		ContainerId:  "FakeContainerId",
+		ListenAddress: &sensor.NetworkAddress{
+			Port: 80,
+		},
+		Originator: &storage.NetworkProcessUniqueKey{
+			ProcessName:         "socat",
+			ProcessExecFilePath: "/usr/bin/socat",
+			ProcessArgs:         "port: 80",
+		},
+	}
+	closedNetworkEndpoint = &sensor.NetworkEndpoint{
+		SocketFamily:   sensor.SocketFamily_SOCKET_FAMILY_IPV4,
+		Protocol:       storage.L4Protocol_L4_PROTOCOL_TCP,
+		ContainerId:    "FakeContainerId",
+		CloseTimestamp: protoconv.ConvertTimeToTimestamp(time.Now()),
+		ListenAddress: &sensor.NetworkAddress{
+			Port: 80,
+		},
+		Originator: &storage.NetworkProcessUniqueKey{
+			ProcessName:         "socat",
+			ProcessExecFilePath: "/usr/bin/socat",
+			ProcessArgs:         "port: 80",
+		},
+	}
+)
+
+func TestNetworkflowManager(t *testing.T) {
+	suite.Run(t, new(NetworkflowManagerTestSuite))
+}
+
+type NetworkflowManagerTestSuite struct {
+	suite.Suite
+}
+
+func (suite *NetworkflowManagerTestSuite) TestAddNothing() {
+	h := hostConnections{}
+	h.endpoints = make(map[containerEndpoint]*connStatus)
+
+	networkInfo := &sensor.NetworkConnectionInfo{}
+	nowTimestamp := timestamp.Now()
+	var sequenceID int64
+	err := h.Process(networkInfo, nowTimestamp, sequenceID)
+	suite.NoError(err)
+	suite.Len(h.endpoints, 0)
+}
+
+func (suite *NetworkflowManagerTestSuite) TestAddOpen() {
+	h := hostConnections{}
+	h.endpoints = make(map[containerEndpoint]*connStatus)
+
+	networkInfo := &sensor.NetworkConnectionInfo{
+		UpdatedEndpoints: []*sensor.NetworkEndpoint{openNetworkEndpoint},
+	}
+
+	nowTimestamp := timestamp.Now()
+	var sequenceID int64
+	h.connectionsSequenceID = sequenceID
+	err := h.Process(networkInfo, nowTimestamp, sequenceID)
+	suite.NoError(err)
+	suite.Len(h.endpoints, 1)
+}
+
+func (suite *NetworkflowManagerTestSuite) TestAddOpenAndClosed() {
```

Comment:
This unit test is failing due to a bug in getUpdatedContainerEndpoints. Specifically the issue is that ep is used as a key in a map, but processKey is a pointer, which can take different values, even if the process is the same. This was fixed by storing the process pointers in a map, and making sure that identical processes use the same pointer.

Reply:
Should now be passing.